airflow/providers/amazon/aws/hooks/s3.py

```diff
@@ -263,6 +281,10 @@ def list_keys(
         delimiter: Optional[str] = None,
         page_size: Optional[int] = None,
         max_items: Optional[int] = None,
+        start_after_key: Optional[str] = None,
+        from_datetime: Optional[DateTime] = None,
+        to_datetime: Optional[DateTime] = None,
```

Comment:
I don’t think this needs to take `pendulum.DateTime`. Normal `datetime.datetime` works equally well (and is compatible with `pendulum.DateTime`).

Reply:
`LastModified` in the key returned by boto3 is timezone aware and comparison with DateTime specified by the user would create `TypeError: can't compare offset-naive and offset-aware datetimes`.

Reply:
Native datetime can be timezone-aware as well; on the other hand, using `pendulum.DateTime` still _does not_ guarantee the instance is timezone-aware.